### PR TITLE
[FIX] Type narrowing related bugs

### PR DESCRIPTION
**Pull request by ayeshLK**
## Examples
N/A

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/http_commons.bal [hunk 1]
```diff
@@ -114,13 +118,16 @@ isolated function buildResponse(ResponseMessage message) returns Response|Listen
         response.setByteStream(message);
     } else if (message is mime:Entity[]) {
         response.setBodyParts(message);
-    } else {
+    } else if (message is anydata) {
```
Comment (chamil321) [inline ballerina/http_commons.bal:121]:
```suggestion
    } else if message is anydata {
```

File: ballerina/http_commons.bal [hunk 2]
```diff
@@ -57,7 +57,7 @@ isolated function buildRequest(RequestMessage message, string? mediaType) return
         request.setByteStream(message);
     } else if (message is mime:Entity[]) {
         request.setBodyParts(message);
-    } else {
+    } else if (message is anydata) {
```
Comment (chamil321) [inline ballerina/http_commons.bal:60]:
```suggestion
    } else if message is anydata {
```

File: ballerina/http_commons.bal [hunk 3]
```diff
@@ -114,13 +118,16 @@ isolated function buildResponse(ResponseMessage message) returns Response|Listen
...
         var result = trap val:toJson(message);
         if (result is error) {
             return error InitializingOutboundResponseError("json conversion error: " + result.message(), result);
         } else {
             response.setJsonPayload(result);
         }
+    } else {
+        string errorMsg = "invalid response body type. expected one of the types: string|xml|json|byte[]|Entity[]|stream<byte[],io:Error?>";
```
Comment (chamil321) [inline ballerina/http_commons.bal:129]:
```suggestion
        string errorMsg = "invalid response body type. expected one of the types: Response|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```

File: ballerina/http_commons.bal [hunk 4]
```diff
@@ -68,6 +68,10 @@ isolated function buildRequest(RequestMessage message, string? mediaType) return
                 request.setJsonPayload(payload);
             }
         }
+
+    } else {
+        string errorMsg = "invalid request body type. expected one of the types: string|xml|json|byte[]|Entity[]|stream<byte[],io:Error?>";
```
Comment (chamil321) [inline ballerina/http_commons.bal:73]:
```suggestion
        string errorMsg = "invalid request body type. expected one of the types: Request|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```

File: ballerina/http_request.bal [hunk 5]
```diff
@@ -552,8 +552,11 @@ public class Request {
             self.setJsonPayload(payload);
         } else if (payload is stream<byte[], io:Error?>) {
             self.setByteStream(payload);
-        } else {
+        } else if (payload is mime:Entity[]) {
             self.setBodyParts(payload);
+        } else {
+            panic error Error("invalid entity body type." +
+                "expected one of the types: string|xml|json|byte[]|Entity[]|stream<byte[],io:Error?>");
```
Comment (chamil321) [inline ballerina/http_request.bal:559]:
```suggestion
                "expected one of the types: string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>");
```

File: ballerina/http_response.bal [hunk 6]
```diff
@@ -488,8 +488,11 @@ public class Response {
             self.setJsonPayload(payload);
         } else if (payload is stream<byte[], io:Error?>) {
             self.setByteStream(payload);
-        } else {
+        } else if (payload is mime:Entity[]) {
             self.setBodyParts(payload);
+        } else {
+            panic error Error("invalid entity body type." +
+                "expected one of the types: string|xml|json|byte[]|Entity[]|stream<byte[],io:Error?>");
```
Comment (chamil321) [inline ballerina/http_response.bal:495]:
```suggestion
                "expected one of the types: string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>");
```

File: ballerina/http_commons.bal [hunk 7]
```diff
@@ -68,6 +68,10 @@ isolated function buildRequest(RequestMessage message, string? mediaType) return
                 request.setJsonPayload(payload);
             }
         }
+
+    } else {
+        string errorMsg = "invalid request body type. expected one of the types: Request|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```
Comment (ayeshLK) [inline ballerina/http_commons.bal:73]:
```suggestion
        string errorMsg = "invalid request body type. expected one of the types: http:Request|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```

File: ballerina/http_commons.bal [hunk 8]
```diff
@@ -114,13 +118,16 @@ isolated function buildResponse(ResponseMessage message) returns Response|Listen
...
-    } else {
+    } else if message is anydata {
         var result = trap val:toJson(message);
         if (result is error) {
             return error InitializingOutboundResponseError("json conversion error: " + result.message(), result);
         } else {
             response.setJsonPayload(result);
         }
+    } else {
+        string errorMsg = "invalid response body type. expected one of the types: Response|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```
Comment (ayeshLK) [inline ballerina/http_commons.bal:129]:
```suggestion
        string errorMsg = "invalid response body type. expected one of the types: http:Response|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```

File: ballerina/http_commons.bal [hunk 9]
```diff
@@ -68,6 +68,10 @@ isolated function buildRequest(RequestMessage message, string? mediaType) return
                 request.setJsonPayload(payload);
             }
         }
+
+    } else {
+        string errorMsg = "invalid request body type. expected one of the types: http:Request|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```
Comment (chamil321) [inline ballerina/http_commons.bal:73]:
```suggestion
        string errorMsg = "invalid request body type. expected one of the types: http:Request|xml|json|table<map<json>>|(map<json>|table<map<json>>)[]|mime:Entity[]|stream<byte[], io:Error?>";
```

File: ballerina/http_commons.bal [hunk 10]
```diff
@@ -114,13 +118,16 @@ isolated function buildResponse(ResponseMessage message) returns Response|Listen
...
-    } else {
+    } else if message is anydata {
         var result = trap val:toJson(message);
         if (result is error) {
             return error InitializingOutboundResponseError("json conversion error: " + result.message(), result);
         } else {
             response.setJsonPayload(result);
         }
+    } else {
+        string errorMsg = "invalid response body type. expected one of the types: http:Response|string|xml|json|byte[]|mime:Entity[]|stream<byte[],io:Error?>|()";
```
Comment (chamil321) [inline ballerina/http_commons.bal:129]:
```suggestion
        string errorMsg = "invalid response body type. expected one of the types: http:Response|xml|json|table<map<json>>|(map<json>|table<map<json>>)[]|mime:Entity[]|stream<byte[], io:Error?>";
```

**Comment by chamil321** [comment]
@ayeshLK let's fix the conflict as well

**Pull request merged**